### PR TITLE
Simplify the helm release tool (#6734)

### PR DESCRIPTION
**Pull request by thbkrkr**
Backport the following commit to `2.8`:
- #6734

### CODE REVIEW

File: .buildkite/pipeline-release-helm.yml [hunk 1]
```diff
@@ -36,7 +38,9 @@ steps:
         commands:
           - buildkite-agent artifact download "bin/releaser" /usr/local/
           - chmod u+x /usr/local/bin/releaser
-          - releaser --env=dev --excludes=eck-operator --dry-run=\$DRY_RUN
+          - releaser --env=dev --charts-dir=deploy/eck-stack --dry-run=\$DRY_RUN
```
Comment (barkbay) [inline .buildkite/pipeline-release-helm.yml:41]: Do we have to allow to re-release non SNAPSHOT charts in the dev Helm repo in we want to test this?

File: deploy/README.md [hunk 2]
```diff
@@ -6,20 +6,27 @@ This directory contains the Helm chart for deploying the ECK operator, and chart
...
+```sh
+helm repo add elastic https://helm.elastic.co
+helm repo update
+```
+
 Install the CRDs and deploy the operator with cluster-wide permissions to manage all namespaces.
 
 ```sh
-helm install elastic-operator eck-operator -n elastic-system --create-namespace 
+helm install elastic-operator elastic/eck-operator -n elastic-system --create-namespace
```
Comment (barkbay) [inline deploy/README.md:19]: See my comment here: https://github.com/elastic/cloud-on-k8s/pull/6746#discussion_r1182520072:

> I think this is done on purpose when you want to install the operator using the Helm Chart in this directory, not from the elastic repository.

File: .buildkite/pipeline-release-helm.yml (hunk 1 above)
Comment (thbkrkr) [reply .buildkite/pipeline-release-helm.yml:41]: Yes but I would do it in another PR. I can work on it.

Comment (barkbay) [reply .buildkite/pipeline-release-helm.yml:41]: > Yes but I would do it in another PR. I can work on it.

Comment (barkbay) [reply .buildkite/pipeline-release-helm.yml:41]: > Yes but I would do it in another PR. I can work on it.

👍 , absolutely, sorry for not being clear. It was to be sure that this was required to test this change "for real".

File: deploy/README.md (hunk 2 above)
Comment (thbkrkr) [reply deploy/README.md:19]: Sorry, I missed it. I think we should use #6746 to fix this, and not modify this backport.

Comment (barkbay) [reply deploy/README.md:19]: > Sorry, I missed it. I think we should use #6746 to fix this, and not modify this backport.

SGTM, I added the `v2.8.0` label to https://github.com/elastic/cloud-on-k8s/pull/6746

**Review by barkbay — APPROVED** [review]
LGTM!

**Pull request merged**